src/main/java/org/carlmontrobotics/subsystems/Arm.java

```diff
@@ -28,21 +45,35 @@ public Arm() {
       */
     }
 	
-    public void setArmPos() {
+    public void setArmGoal(double targetPosition, double targetVelocity) {
       //Sets arm to the optimal angle for amp, speaker and Ground intake | used to score in amp
       //these values are in constants
-      //pass in where scorign and use switch statement
-    
-
+      //pass in where scorign and use switch statement to alternate between each angle needed
+      targetPosition = getArmClampedGoal(targetPosition);
+      
     }
     
-    public void setSpeed() {
-      //move the arm around based off the right joystick movement on the manipulator joystick
-      //use the trapezoid thingy from robot code 2023
+    public double getArmClampedGoal(double goal) {
+      //Find the limits of the arm. Used to move it and ensure that the arm does not move past the amount
+      return MathUtil.clamp(MathUtil.inputModulus(goal, Constants.Arm.ARM_DICONT_RAD, Constants.Arm.ARM_DICONT_RAD + 2 * Math.PI), Constants.Arm.LOWER_ANGLE, Constants.Arm.UPPER_ANGLE);
     }
-    
+
     @Override
     public void periodic() {
-     // run moveArm passing in controller here
+  
 		}
+
+    public void driveArm(TrapezoidProfile.State state){
+        // set voltage to armMotor
+        double kgv = getKg();
+        double armFeedVolts = kgv * getCoM().getAngle().getCos() + armFeed.calculate(state.velocity, 0);
+        double armPIDVolts = armPID.calculate(getArmPos(), state.position);
+        if ((getArmPos() > ARM_UPPER_LIMIT_RAD && state.velocity > 0) || 
+            (getArmPos() < ARM_LOWER_LIMIT_RAD && state.velocity < 0)) {
+              forbFlag = true;  
+            armFeedVolts = kgv * getCoM().getAngle().getCos() + armFeed.calculate(0, 0);
+        }
+        double volts = armFeedVolts + armPIDVolts;
+        armMotor.setVoltage(volts);
```

Comment:
This code works for an arm with a wrist joint that is controlled by a different motor. This code DOES NOT work for an arm without a wrist joint. Don't randomly copy paste code from previous years if you don't understand if it works. Also, compilation errors.